source/common/protobuf/well_known.h

```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+namespace Envoy {
+namespace ProtobufWellKnown {
+
+// Used by VersionConverter to track the original type of an upgraded message.
+constexpr uint32_t OriginalTypeFieldNumber = 100000;
```

Comment:
Is this generally how this type of thing is done? Is there any way to reserve this field number just to make sure it doesn't get used otherwise? Probably not?

Reply:
Yeah, I don't see any easy way. As with extensions, we can use the top few MSBs to make it quasi-unique, wil ldo that.